### DISCUSSION TITLE
Support bit string operators of Postgresql

### DISCUSSION THREAD

**Discussion by hurcy** [sqlalchemy/sqlalchemy#5954] [Usage Questions] [answered]
**Describe your question**
Hi,

I have bit(10) column and want to update it using bitwise operator.
e.g. 
```
from sqlalchemy.dialects.postgresql import BIT
Mytable = db.Table(
  db.Column(ColumnLabel.EVENT_TYPE.value, BIT(10), nullable=False)
)

q1 = db.session.query(
    (Mytable.c.event_type | cast(8, BIT(10))
)

q2 = db.session.query(
    or_(Mytable.c.event_type,  cast(8, BIT(10))
)
```

Both queries compiled to `select event_type OR CAST(8 AS BIT(10))`, which is not correct. 
Could you tell me how to compile those queries with `|` operator?

**Example - please use the [Minimal, Complete, and Verifiable](http://stackoverflow.com/help/mcve) guidelines if possible**

**Complete stack trace, if applicable**

**Versions**
 - OS: macOS Big Sur 11.1
 - Python: 3.8.6
 - SQLAlchemy: 1.3.22
 - Database: Postgresql 12
 - DBAPI: psycopg2

**Additional context**

**Useful links**
- [Get Support](https://www.sqlalchemy.org/support.html) on the website
- The [documentation](https://docs.sqlalchemy.org/en/latest/) website
- The [UsageRecipes](https://github.com/sqlalchemy/sqlalchemy/wiki/UsageRecipes) wiki
- [Stack Overflow](https://stackoverflow.com/questions/tagged/sqlalchemy) tag
- SQLAlchemy [Google group](http://groups.google.com/group/sqlalchemy)
- [Gitter](https://gitter.im/sqlalchemy/community) chat

**Have a nice day!**

**Comment by zzzeek** [comment, marked as answer]
hi -

SQLAlchemy [implements the bitwise | and & operator as SQL OR and AND](https://docs.sqlalchemy.org/en/13/core/sqlelement.html?highlight=#sqlalchemy.sql.expression.and_).    To get the actual "bitwise" operators for this uncommon database type, you would need to use [op()](https://docs.sqlalchemy.org/en/13/core/sqlelement.html#sqlalchemy.sql.operators.Operators.op):

```python
from sqlalchemy import cast
from sqlalchemy import Column
from sqlalchemy import MetaData
from sqlalchemy import Table
from sqlalchemy.dialects.postgresql import BIT

Mytable = Table("table", MetaData(), Column("col", BIT(10), nullable=False))

print(Mytable.c.col.op("|")(cast(8, BIT(10))))

```
